generators/lua/math.js

```diff
@@ -196,7 +196,11 @@ Blockly.Lua['math_number_property'] = function(block) {
           
           const divisor = Blockly.Lua.valueToCode(block, 'DIVISOR',
               Blockly.Lua.ORDER_MULTIPLICATIVE) || '0';
-      
+           // If 'divisor' is some code that evals to 0, Lua will produce a nan.
+           // Let's produce nil if we can determine this at compile-time.
+           if (!divisor || divisor == '0') {
```

Comment:
By using `|| '0'` in the initialization of `divisor` above, `!divisor` becomes unnecessary here.